### DISCUSSION TITLE
Versions of modules.

### DISCUSSION THREAD

**Discussion by lzof** [flamingock/mongock#11] [Q&A] [answered]
At first, thanks very much for providing such a good tool.

But it looks like we have com.github.cloudyrock.mongock:mongodb-v3-driver:**4.3.2** on maven now.
While we only see com.github.cloudyrock.mongock:mongock-standalone:**4.1.19** on maven.

How could you manage the versions of modules and do the upgrading?
How can we select the corresponding versions between modules?

Looking forward to hearing from you.

Thanks.

**Comment by dieppa** [comment, marked as answer]
Hello @lzof ,

that's a good question. Short answer is that you shouldn't care, as you should be using the mongock bom, instead of importing the indicating the versions directly for each library. Please take a look to the [documentation](https://www.mongock.io/quick-start).

The long answer it's complex to explain, depending when I start. To make it simpler, we have mongock and mongock-core, which is several versiones behind mongock. In the last release we decided to move the runners, initially in Mongock, to mongock-core, keeping the same names for the artifact to make it transparent to the users. What we didn't realises was that mongock-core is in version 3, the runners will have that version, but we have already release higher versions for the same artifact.
As users should be using the mongock bom, we didn't really worry too much, in the next release we just increase the core's version to match the current mongock version and fixed.

So I hope you have some idea of why is that, in summary was a mistake that will be fixed in the next release, but keep into account that runners and drivers are independent and the versions won't need to match, the probably won't. But you don't care as that's handled by the mongock bom.

**Reply by lzof** [reply]
Many thanks for the reply. I am using gradle for the dependencies management. And I was not quite familiar with migrating maven projects to gradle.

Anyway, I would like an elegant solution. After research, I found we can import the dependencies in gradle as follows.
Hope we can add it to your cook book for users with gradle based projects, if you think it is fine.

```
dependencies {
  implementation platform("com.github.cloudyrock.mongock:mongock-bom:4.3.2")
  implementation("com.github.cloudyrock.mongock:mongock-standalone")
  implementation("com.github.cloudyrock.mongock:mongodb-v3-driver")
}
```

Thanks for providing such a good tool again. Much appreciated.

**Reply by dieppa** [reply]
I am glad I was able to resolve your doubt. We'll definitely use that in tutorial. 
Thanks a lot
